mission-vision-values.md

```diff
@@ -0,0 +1,99 @@
+
+Excellent observability requires high-quality telemetry, as well as the
+performant, consistent instrumentation that makes it possible.
+
+Telemetry in this context is the firehose of raw observational data streaming
+out of software applications, and while “high-quality telemetry” may be a
+requirement for excellent observability, it’s still unreasonable and
+unrealistic to expect software developers to add or maintain the necessary
+instrumentation on their own. That is a massive undertaking, and in practice
+it’s not “just code” – there is also necessary alignment around protocols and
+semantic conventions for tags, attributes, and other metadata to consider.
+
+So how do we get high-quality, turnkey telemetry without a massive,
+unsustainable engineering effort? This is where OpenTelemetry comes in. To
+achieve our vision, OpenTelemetry sees five key opportunities, listed here:
+
+**Opportunity I: Telemetry should be easy.** With OpenTelemetry we want
+high-quality telemetry to be easy, especially for end-users. That means that
+OpenTelemetry should have fast time-to-value, set reasonable defaults yet allow
+for customization, and pair with excellent documentation and a top-tier overall
+developer experience.
+
+**Opportunity II: Telemetry should be universal.** Telemetry protocols and
+conventions should be unified across languages and signal types (tracing,
+metrics, logging, etc), not divergent or siloed. This means that OpenTelemetry
+aspires to find technical solutions that work consistently, both locally and
+globally.
+
+**Opportunity III: Telemetry should be vendor-neutral.** For decades, monitoring
+and observability vendors have created differentiation – and lock-in – through
+agents that automatically gather useful telemetry from a wide integration
+surface. While convenient, this lock-in is limiting for customers and inhibits
+innovation by tightly coupling telemetry collection and telemetry storage and
+analysis. With OpenTelemetry, we strive to provide a level playing field for
+all observability providers, avoiding lock-in for any vendor and interoperating
+with other OSS projects in the telemetry and observability ecosystem.
+
+**Opportunity IV: Telemetry should be loosely coupled.** OpenTelemetry end-users
+should be able to pick and choose from the pieces they want without bringing in
+the rest of the project. To enable this, OpenTelemetry’s software architecture
+is decoupled wherever possible. As a corollary, this also means that
+OpenTelemetry does not want to “pick winners” when it comes to particular
+projects or technologies: where possible, we prefer to create optionality for
+our end-users.
+
+**Opportunity V: Telemetry should be built-in.** Historically, telemetry was
+something developers integrated manually or via post-compilation agents.
+OpenTelemetry believes that high-quality telemetry can be built in to the
+entire software stack – just like testing has been for decades.
```

Comment:
I find the comparison with testing a bit confusing, not sure if it helps in any way (and it surely distract me since as I read this I try to find the similarities and differences between testing and observability :-) ). Perhaps drop the reference to testing.

Reply:
disagree - it's an analogy to writing comments, or writing unit tests - that it should be easy to do so, so that it shifts left.

Reply:
(I eagerly wait additional opinions on this matter :)

Reply:
I see what Liz is saying, but the text does not read like that

Reply:
the phrasing _I_ like to use that we can steal is that writing code without comments or tests would be unthinkable, so code without telemetry should similarly be unthinkable.

Reply:
> writing code without comments or tests would be unthinkable

I wish this were true, but I am onboard with the sentiment. After all, we're describing the vision.

Reply:
+1 to the reference to testing being somewhat confusing (_testing_ isn't necessarily built-in to JDBC or Express.js). But I do strongly agree with the tenet of observability via OpenTelemetry being built-in to software packages.

Reply:
Is CI a better analogy here?

25 years ago, the expectation was that a library consisted of a tarball and release notes. Today with Github and its ecosystem, we expect that tests run regularly, can be reviewed by contributors and adopters, and are subject to flakiness hygiene.